fast64_internal/oot/actor/properties.py

```diff
@@ -110,22 +164,193 @@ def draw_props(
+                    paramValue = getEvalParamsInt(
+                        curPropValue if not param.type == "Bool" else "1" if curPropValue else "0"
+                    )
+                else:
+                    if param.type == "Type":
+                        typeValue = getEvalParamsInt(curPropValue)
+                    else:
+                        paramValue = 0
+                        if param.type == "ChestContent":
+                            paramValue = ootData.actorData.chestItemByKey[curPropValue].value
+                        elif param.type == "Collectible":
+                            paramValue = ootData.actorData.collectibleItemsByKey[curPropValue].value
+                        elif param.type == "Message":
+                            paramValue = ootData.actorData.messageItemsByKey[curPropValue].value
+
+                if "Rot" in target:
+                    typeValue = getEvalParamsInt(getattr(self, getObjName(actor.key, "Type", None, 1)))
+
+                if typeValue is not None and typeValue in param.tiedTypes or len(param.tiedTypes) == 0:
+                    val = ((paramValue if paramValue is not None else -1) & param.mask) >> getShiftFromMask(param.mask)
+                    isInRange = self.isValueInRange(val, param.valueRange[0], param.valueRange[1])
+                    if isInRange and param.type != "Type" and paramValue is not None:
+                        value = getFormattedParams(param.mask, paramValue, param.type == "Bool")
+                        if value is not None:
+                            paramList.append(value)
+
+        if len(paramList) > 0:
+            paramString = " | ".join(val for val in paramList)
+        else:
+            paramString = "0x0"
+
+        if "Rot" in target:
+            typeValue = None
+
+        evalTypeValue = typeValue if typeValue is not None else 0
+        evalParamValue = getEvalParamsInt(paramString)
+
+        if evalTypeValue and evalParamValue and typeValue is not None:
+            paramString = f"(0x{typeValue:04X} | ({paramString}))"
+        elif evalTypeValue and not evalParamValue and typeValue is not None:
+            paramString = f"0x{typeValue:04X}"
+        elif not evalTypeValue and evalParamValue:
+            paramString = f"({paramString})"
+        else:
+            paramString = "0x0"
+
+        return paramString if not self.evalParams else getEvalParams(paramString)
+
+    def draw_params(self, layout: UILayout, objName: str):
+        actor = ootData.actorData.actorsByID[self.actorID]
+        curType = None
+        for param in actor.params:
+            propName = getObjName(actor.key, param.type, param.subType, param.index)
+
+            if param.type == "Type":
+                curType = getEvalParamsInt(getattr(self, propName))
+
+            if curType is not None and curType in param.tiedTypes or param.type == "Type" or len(param.tiedTypes) == 0:
```

Comment:
Make this `elif`? Or maybe add a comment explaining that the first param is always of type `"Type"`, and that there is only one param like this. This code is a little confusing without this context.